components/dashboard/components/NetworkGraph/NetworkGraphView.jsx

```diff
@@ -0,0 +1,623 @@
+
+        // If there are no user interactions for the current user, then return
+        // Only the 'You' node will be shown
+        if (!userInteractions) {
+            return false;
+        }
+
+        /**
+         * The keys of interactionContexts will be the channel name in which interactions
+         * took place. Course interactions will be grouped under a 'course' key.
+         * userInteractions is an array of interactions in which the current user took part in.
+         * They need to be grouped into interaction contexts and then nodes need to be prepared.
+         *     The process:
+         *         1. Create an object with interaction contexts as keys (interactionContexts)
+         *         2. For each interaction context, generate:
+         *                                                  a. a node configuration
+         *                                                  b. a users object with usernames as keys to store
+         *                                                     interaction data for each user in the context
+         *         3. Then, for each user in each interaction context, generate a node configuration
+        */
+        let interactionContexts = {};
+
+        // Get user's interactions in the course context
+        const courseInteractions = userInteractions.filter((interaction) => {
+            return interaction.context === 'course';
+        });
+
+        // If the user has interactions in the course context,
+        // create a node configuration object and interaction context object for the course node
+        if (courseInteractions.length > 0) {
+            const courseContext = this.getCourseContext(courseInteractions);
+            interactionContexts.course = courseContext;
+        }
+
+        // If this user is a member of learning groups AND there are defined learning group types,
+        // then generate interaction context objects for all learning groups and add them to the main
+        // interaction contexts object.
+        if (learningGroupTypes && learningGroupTypes.length > 0 && userLearningGroups && userLearningGroups.length > 0) {
+            const learningGroupContexts = this.getLearningGroupContexts(userInteractions, userLearningGroups, learningGroupTypes);
+
+            interactionContexts = {
+                ...interactionContexts,
+                ...learningGroupContexts,
+            };
+        }
+
+        // For each user in each interaction context, generate a node configuration
+        // Add each user's node as a child of its corresponding interaction context
+        // Add each interaction context's node as a child of the 'You' node
+        Object.keys(interactionContexts).forEach((interactionContext) => {
+            Object.keys(interactionContexts[interactionContext].users).forEach((user) => {
+                // Set configuration for uncontacted group members
+                let {fillOpacity, strokeOpacity} = 1;
```

Comment:
I'm pretty sure this isn't doing what you intended if what you intended was:
```
let fillOpacity = 1;
let strokeOpacity = 1;
```
because it is a destructuring assignment, and the primitive number 1 has no `fillOpacity` or `strokeOpacity` property so those 2 variables are undefined after this statement.
I'm going to fix this.